src/main/java/it/pagopa/ecommerce/commons/domain/v1/TransactionExpired.java

```diff
@@ -51,7 +52,7 @@ public Transaction applyEvent(Object event) {
             case TransactionRefundErrorEvent e ->
                     new TransactionWithRefundError(this.getTransactionAtPreviousState(), e);
             case TransactionRefundRequestedEvent e ->
-                    new TransactionWithRefundRequested(this.getTransactionAtPreviousState(), e);
+                    new TransactionWithRefundRequested((BaseTransactionWithRequestedAuthorization) this.getTransactionAtPreviousState(), e);
```

Comment:
What about making two aggrgeates for expired transactions:
 * `BaseTransactionExpiredNotAuthorized`, inheriting from `BaseTransaction`
 * `BaseTransactionExpired` inheriting from `BaseTransactionWithAuthorizationRequested`

This way we could avoid the cast and make it impossible to get a `TransactionExpired` from an unauthorized transaction

Reply:
Fixed in PR #49
With CANCELLATION_EXPIRED final status TransactionExpired can extends directly `BaseTransactionWithAuthorizationRequested` removing the casting at all.
This can be conceptually linked to the fact that, as for now, `EXPIRED` state is a non-final state where the only plausible state transition is `REFUND_REQUESTED`, meaning the refund process for the transaction to have been start.

Since a transaction can expiry without having requested an authorization at all, for those transactions no refund process has to be performed.

Given the above points there is no meaning for an expired transactions for which no authorization has been requested to go into `EXPIRED` state since no REFUND can be effectively performed.